projs/shadow/shadow-engine/core/inc/shadow/core/jobs/Job.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+#include <atomic>
+#include <cstdint>
+#include <functional>
+
+
+/**
+ * A threaded task/job system.
+ * Can run multiple things in parallel, with automatic dependency ordering.
+ * Create a TaskContext and call the static execute or dispatch methods to run stuff off-thread.
+ */
+namespace SH::Jobs {
+    /**
+     * The data that a task needs in order to handle itself and its relation with other tasks that may be running simultaneously.
+     * Most data relates to task groups, but the idx means this is easier set per task.
+     */
+    struct TaskArguments {
+        uint32_t idx;           // The index of the current task. When using dispatch, this is the iterator index, denoting the nth task that was started.
+        uint32_t group;         // The index of the current task group. When using dispatch, this is the group index, denoting the nth dispatch that was started.
+        uint32_t groupIdx;      // The index of the current task in the group.
+        bool first;             // Whether this is the first task in its' group.
+        bool last;              // Whether this is the last task in its' group.
+        void* data;             // Data shared between a whole group.
+    };
+
+    /**
+     * A task priority.
+     * Higher priority tasks will pause lower priority tasks when all threads are being utilized.
+     */
+    enum class Priority {
+        HIGHEST,        // Default priority.
+        LOW,            // Low priority tasks. Do not pause regular tasks.
+        STREAM,         // Resource streaming, background tasks that never take priority over others
+        SIZE
```

Comment:
This is an `enum class`, not an enum. It can only be accessed by Priority::SIZE. 
This name is inline with the 16 enums in the rx rewrite that have a SIZE field, which makes sense in context:

```
    Shader              shaders[defs::ShaderType::SIZE];
    Texture             texs[defs::TextureType::SIZE];
    InputLayout         layouts[defs::InputLayout::SIZE];
    RasterizerState     rasters[defs::RasterizerState::SIZE];
    DepthStencilState   stencils[defs::DepthStencilState::SIZE];
    BlendState          blends[defs::BlendModes::SIZE];
    GPUBuffer           buffers[defs::BufferType::SIZE];
    Sampler             samplers[defs::SamplerType::SIZE];
```

Reply:
If this is an established name than fine. But I would have preferred if it was more obvious that it was a meta value and not an actual usable member of the enum.